legate/core/runtime.py

```diff
@@ -711,14 +729,389 @@ def record_partition(self, index_space, functor, index_partition):
+                    fusable_intervals.append((start, start+1))
+                    start=start+1
+                    end = start
+        if start<end:
+            fusable_intervals.append((start,end))
+        drint(fusable_intervals)   
+        drint("allFusableOps", results)
+        fusability_exists = reduce(lambda x,y: x or y,[int(op._task_id) in self.validIDs for op in ops])
+        return (fusability_exists, fusable_intervals)
+
+class ValidScalarProducers(FusionConstraint):
+   """Checks all scalar producing are terminal ops"""
+
+class IdenticalProjection(FusionConstraint):
+    """Fusion rule that only ops with identical
+       projection functors can be fused"""
+
+    def apply(self, contexts, runtime, ops, partitioners, strategies):
+        linkset = {}
+        store_to_ops = {}
+        base_window = [0, len(ops)]
+        
+        intervals = []
+        start=0
+        end = len(ops)
+        i=0
+        #for i, op in enumerate(ops):
+        while i<end:
+            op=ops[i]
+            bufferSet = {}
+            # find the set union of input and output buffers for the op
+            for input, part in zip(op._inputs, op._input_parts):
+                if input not in bufferSet:
+                    proj = strategies[i].get_projection(part)
+                    if hasattr(proj, 'part'):
+                        bufferSet[input]=proj
+                        #if input not in linkset:
+                        #    linkset[input] = [i]
+                        #else:
+                        #    linkset[input].append(i)
+            for output, part in zip(op._outputs, op._output_parts):
+                if output not in bufferSet:
+                    proj = strategies[i].get_projection(part)
+                    if hasattr(proj, 'part'):
+                        bufferSet[output]=proj
+                        #if output not in linkset:
+                        #    linkset[output] = [i]
+                        #else:
+                        #    linkset[output].append(i)
+            if i==0: #we only iterate from i==1 onwards
+                i+=1
+                continue
+            # for each op in the union, record its associated transform
+            for buffer in bufferSet.keys():
+                proj = bufferSet[buffer]
+                matrix = proj.part.index_partition.functor.transform.trans
+                if buffer not in store_to_ops:
+                    store_to_ops[buffer] = matrix
+                else: #we see a new projection for the same buffer
+                    if not np.array_equal(matrix, store_to_ops[buffer]):
+                        intervals.append((start, i))
+                        start=i
+                        i=start+1
+                        store_to_ops={}
+                        continue
+            i+=1
+        if start<end:
+            intervals.append((start,end))
+        #print("proj", intervals)
+        return True, intervals
+         
```

Comment:
As stated, the above constraint isn't particularly cheap....and it doesn't seem to ever be "used" I will look into the perf, but I suspect the `get_projection` method is the culprit here